src/scripts/add_claim_subject_column.js

```diff
@@ -0,0 +1,38 @@
+const { createConnection, TableColumn } = require('typeorm');
+const { JWT } = require('@ew-did-registry/jwt');
+const { Keys } = require('@ew-did-registry/keys');
+const { Claim } = require('../../dist/modules/claim/claim.entity.js');
+
+(async function () {
+  const connection = await createConnection({
+    // had to copy from orgmconfig because typeorm doesn't detect postgres driver
+    type: "postgres",
+    host: "localhost",
+    port: 5432,
+    username: "postgres",
+    password: "password",
+    database: "dev",
+    "entities": [
+      "dist/**/*.entity.js"
+    ]
+  });
+  const claimsRepository = connection.getRepository(Claim);
+  let claims = await claimsRepository.find();
+
+  const queryRunner = connection.createQueryRunner();
+  if (!(await queryRunner.hasColumn("claim", "subject"))) {
+    await queryRunner.addColumn("claim", new TableColumn({ name: 'subject', type: 'varchar', default: "''" }));
+  }
+  
+  const jwt = new JWT(new Keys());
+  for await (const claim of claims) {
+    const payload = jwt.decode(claim.token);
+    let subject = payload.sub;
+    if (!subject || subject.length === 0 || !subject.startsWith("did")) {
+      subject = payload.iss;
+      await claimsRepository.update(claim.id, { ...claim, subject });
```

Comment:
Shouldn't this be outside of the `if` branch? For this script, we always want to update the claim `subject` as it wasn't set at all before?

Reply:
My bad